// Copyright lowRISC contributors (OpenTitan project).
// Licensed under the Apache License, Version 2.0, see LICENSE for details.
// SPDX-License-Identifier: Apache-2.0


// Note: the following definitions were taken directly from the Tink library
// library with some modifications, and in particular from the file:
// https://github.com/google/tink/blob/master/proto/common.proto
// The copyright is preserved below:
// -----------------------------------------------------------------------------
// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// -----------------------------------------------------------------------------

// Definitions for common cryptographic enum types.
syntax = "proto3";

package crypto.common;
option go_package = "common_go_pb";

enum EllipticCurveType {
  ELLIPTIC_CURVE_TYPE_UNSPECIFIED = 0;
  ELLIPTIC_CURVE_TYPE_NIST_P256 = 2;
  ELLIPTIC_CURVE_TYPE_NIST_P384 = 3;
  ELLIPTIC_CURVE_TYPE_NIST_P521 = 4;
  ELLIPTIC_CURVE_TYPE_CURVE25519 = 5;
}

enum EcPointFormat {
  EC_POINT_FORMAT_UNSPECIFIED = 0;
  EC_POINT_FORMAT_UNCOMPRESSED = 1;
}

enum HashType {
  HASH_TYPE_UNSPECIFIED = 0;
  HASH_TYPE_SHA384 = 2;
  HASH_TYPE_SHA256 = 3;
  HASH_TYPE_SHA512 = 4;
  HASH_TYPE_SHA224 = 5;
}
